src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionProposal.java

```diff
@@ -155,12 +176,57 @@ private void updateCompletionItemFromResolved() {
         this.item = resolved;
     }
 
+    @NotNull
+    private String adjustSnippetContent(@NotNull String snippetContent) {
```

Comment:
Hopefully this method is pretty self-explanatory, but basically if client config says not to use editor templates for invocation arg-only completion snippets, see if the snippet contains what appears to be an invocation arg list with template variables, and if so **and** if there are no non-end variables outside of that invocation arg list, just update the snippet to be `<beforeArgList>($0)<afterArgList>` with any other `$0` removed. That way when the snippet is parsed into a template, it will just result in the caret being moved into the invocation arg parens.